### ISSUE TITLE
Make Sure that the Exported Models are Compressed and Tarred

### ISSUE THREAD

**Issue by humblenginr** [meshery/meshery#11605]
### Current Behavior
Exported Models and Designs are only tarred, but not compressed. 

### Desired Behavior
Make sure all the exported Designs and Models are compressed and tarred. 

### Acceptance Tests

Make sure the following user flows are updated:
- [ ] Mesheryctl Model Export
- [ ] Mesheryctl Design Export
- [ ] Meshery UI Model Export 
- [ ] Meshery UI Design Export 

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

**Comment by leecalcote** [comment]
@humblenginr progress?

**Comment by TraXIcoN** [comment]
Hey @leecalcote is this issue open for work?

**Comment by BasavarajBankolli** [comment]
Hi @leecalcote  

I’ve implemented gzip compression for exported Models and Designs on top of existing tar packaging.

### Summary of changes:
- Ensured `.tar.gz` output for OCI-based exports
- Added gzip compression layer without modifying existing packaging logic
- Updated response headers accordingly

### Validation:
- Verified archive integrity using `tar -tzf`
- Successfully extracted archives using `tar -xvzf`
- Confirmed reduced file size compared to `.tar`
- Observed correct OCI artifact structure inside the archive

Note: The CLI currently does not preserve the `.tar.gz` filename extension, but the server response and archive content are correct.

I have opened a PR, Feedback is welcome!

Thanks

**Comment by leecalcote** [comment]
Thanks, @BasavarajBankolli. We'll need to ensure that all of Meshery's components and user flows are compatible with this change:

- verify import of .tar.gz models via Meshery UI
- verify import of .tar.gz models via Meshery CLI
- verify export of .tar.gz models via Meshery UI
- verify export of .tar.gz models via Meshery CLI
- verify https://meshery.io/catalog

**Comment by BasavarajBankolli** [comment]
@leecalcote 

Thanks for the review and guidance!

I’ve now validated the requested scenarios:

- Verified `.tar.gz` export via Meshery CLI and UI
- Confirmed archive structure and integrity
- Checked compatibility with catalog models (no regressions)

Regarding import:
- `.tar.gz` OCI artifacts are currently not supported by Meshery CLI/UI import flows
- Supported formats include Kubernetes manifest, Helm chart, Docker Compose, and Design YAML
- This is consistent with existing behavior and not introduced by this PR

I’ve documented these findings in the PR.

Happy to take up `.tar.gz` import support as a follow-up enhancement if needed.

**Comment by BasavarajBankolli** [comment]
Follow-up idea: Add support for importing `.tar.gz` OCI artifacts by:
- Extracting archive contents
- Detecting contained format (OCI / YAML)
- Integrating with existing import pipeline

Happy to work on this as a separate issue.

**Comment by yi-nuo426** [comment]
@BasavarajBankolli, no. These concerns aren't to be dealt with separately. All of these concerns (and maybe others that haven't been written into the list) MUST be addressed in order for your PR to merge. The are not separate, but an interlinked system.
